tests/externs.el

```diff
@@ -59,3 +59,69 @@
 ;;;TODO
 ;; ensure that any symbols bound with the import syntax are added to the extern list
 ;; ensure that any symbols bound with the export syntax exist in the file scope
+
+;; Test `/*jslint*/` declarations
+
+(ert-deftest js2-finds-jslint-declaration-externs ()
+  (with-temp-buffer
+    (insert "/*jslint browser: true, node: true*/")
+    (js2-mode)
+    (should (equal (delete-dups (sort (js2-get-jslint-declaration-externs) 'string<))
+                   (delete-dups (sort (append (cdr (assoc "browser" js2-jslint-declaration-externs))
+                                              (cdr (assoc "node" js2-jslint-declaration-externs))
+                                              nil)
+                                      'string<)))))
+
+  (ert-deftest js2-no-jslint-declaration-externs-without-keyword ()
+    (with-temp-buffer
+      (insert "/* browser: true, node: true*/")
+      (js2-mode)
+      (should (null (js2-get-jslint-declaration-externs))))))
+
+(ert-deftest js2-no-jslint-declaration-externs-for-nonexistent-env ()
+  (with-temp-buffer
+    (insert "/*jslint nonexistent: true*/")
+    (js2-mode)
+    (should (null (js2-get-jslint-declaration-externs)))))
+
+(ert-deftest js2-finding-jslint-declaration-externs-ignores-nonexistent ()
+  (with-temp-buffer
+    (insert "/*jslint es6: true, nonexistent: true, couch: true*/")
+    (js2-mode)
+    (should (equal (delete-dups (sort (js2-get-jslint-declaration-externs) 'string<))
+                   (delete-dups (sort (append (cdr (assoc "es6" js2-jslint-declaration-externs))
+                                              (cdr (assoc "couch" js2-jslint-declaration-externs))
```

Comment:
> Do you want me to revert to the previous method using member?

Yeah, I think it was fine. Checking for one member from each of the lists seems reliable enough. If any of the lists is missing elements, it would be hard to catch with a test anyway (or the test would become really noisy and thus unreadable).